coreui/src/main/java/com/alexafit/coreui/components/textfield/TextFieldWithIcon.kt

```diff
@@ -1,114 +1,78 @@
     visualTransformation: VisualTransformation = VisualTransformation.None,
     onValueChange: (String) -> Unit,
-    onFocusChanged: (FocusState) -> Unit
+    onFocusChanged: (FocusState) -> Unit,
+    placeholder: @Composable (() -> Unit)? = null,
+    icon: @Composable (() -> Unit)? = null
 ) {
     val spacing = LocalSpacing.current
-
-    Box(
+    TextField(
+        value = text,
+        onValueChange = onValueChange,
         modifier = modifier
-    ) {
-        BasicTextField(
-            value = text,
-            onValueChange = onValueChange,
-            singleLine = true,
-            keyboardActions = KeyboardActions(
-                onDone = {
-                    defaultKeyboardAction(imeAction = keyboardImeAction)
-                }
-            ),
-            keyboardOptions = KeyboardOptions(
-                imeAction = keyboardImeAction,
-                keyboardType = keyboardType
-            ),
-            modifier = Modifier
-                .clip(RoundedCornerShape(spacing.cornerShapeMedium))
-                .shadow(
-                    elevation = spacing.shadowSmall,
-                    shape = RoundedCornerShape(spacing.cornerShapeMedium)
-                )
-                .background(
-                    color = MaterialTheme.colors.secondaryVariant
-                )
-                .fillMaxWidth()
-                .padding(spacing.spaceMedium)
-                .padding(end = spacing.spaceExtraLarge)
-                .onFocusChanged { onFocusChanged(it) },
-            textStyle = MaterialTheme.typography.body1.copy(textAlign = TextAlign.Start),
-            visualTransformation = visualTransformation
-        )
-        if (text.isEmpty()) {
-            Text(
-                text = hint,
-                style = MaterialTheme.typography.body1,
-                fontWeight = FontWeight.Light,
-                color = MaterialTheme.colors.onSurface,
-                modifier = Modifier
-                    .align(Alignment.CenterStart)
-                    .padding(spacing.spaceMedium)
+            .shadow(
+                elevation = spacing.shadowSmall,
+                shape = RoundedCornerShape(spacing.cornerShapeMedium)
+            )
+            .background(
+                color = MaterialTheme.colors.secondaryVariant
             )
-        }
-        iconImage?.let {
-            IconButton(
-                onClick = { iconOnClick() },
-                modifier = Modifier
-                    .align(Alignment.CenterEnd)
-                    .height(IntrinsicSize.Min)
-                    .width(IntrinsicSize.Min)
-            ) {
-                Icon(
-                    painter = painterResource(id = it),
-                    contentDescription = stringResource(
-                        id = iconContentDescription ?: R.string.content_desc_image_vector
-                    ),
-                    tint = iconTint
-                )
+            .fillMaxWidth()
```

Comment:
Here, you're forcing this reusable composable to always span the whole width, no matter where you place it. Imagine you'd want to place 2 of them next to each other. That wouldn't work here. I recommend to not hardcode behavior with `fillMaxWidth/Height()` and instead set that on the modifier parameter passed.